harness/determined/pytorch/_callback.py

```diff
@@ -16,28 +16,32 @@ class PyTorchCallback:
 
     .. warning::
         If distributed training is enabled, every GPU will execute a copy of this callback
-        (except for :meth:`on_validation_step_end` and :meth:`on_checkpoint_end`).  To
+        (except for :meth:`on_validation_end` and :meth:`on_checkpoint_end`).  To
         configure a callback implementation to execute on a subset of GPUs, please condition
         your implementation on ``trial.context.distributed.get_rank()``.
     """
 
-    def on_train_step_start(self, step_id: int) -> None:
+    def on_batch_start(self, batch_idx: int) -> None:
         """
-        Run before every training step begins.
+        Run before every batch is trained.
         """
         pass
 
-    def on_train_step_end(self, step_id: int, metrics: Dict[str, Any]) -> None:
+    def on_batch_end(self, batch_idx: int, metrics: Dict[str, Any]) -> None:
         """
-        Run after every training step ends.
+        Run after every batch is trained.
+        """
+        pass
 
-        .. warning::
-            If distributed training is enabled, every GPU will execute a copy of
-            this callback at the end of every training step. If
-            ``optimizations.average_training_metrics`` is enabled, then the
-            ``metrics`` will be averaged across all GPUs before the callback is
-            executed.  If ``optimizations.average_training_metrics`` is
-            disabled, then the ``metrics`` will be local to the GPU.
+    def on_epoch_start(self, epoch_idx: int) -> None:
+        """
+        Run before every epoch begins.
+        """
+        pass
+
+    def on_epoch_end(self, epoch_idx: int, metrics: Dict[str, Any]) -> None:
+        """
+        Run after every epoch ends.
```

Comment:
blocking: We should be more descriptive here about the timing. Often `epoch_end` is considered to be the end of training and evaluating on a full dataset, which is not the case for us. Might be even worth renaming this callback to `on_train_epoch_end()`.

Reply:
hadn't thought that would be the expected behavior. thanks.